src/components/views/rooms/SendMessageComposer.js

```diff
@@ -74,7 +75,9 @@ export function createMessageContent(model, permalinkCreator, replyToEvent) {
         msgtype: isEmote ? "m.emote" : "m.text",
         body: body,
     };
-    const formattedBody = htmlSerializeIfNeeded(model, {forceHTML: !!replyToEvent});
+    const formattedBody = SettingsStore.getValue("enableMarkdown") &&
```

Comment:
I'm relatively sure moving this to `createMessageContent` would break Markdown-formatted messages by others, hence its inception(s) in the composer components.

Reply:
Doubtful given messages are not markdown over the wire, the sender converts them into HTML and the other ends just renders it.